tests/phpunit/api/v3/ContactTest.php

```diff
@@ -1326,6 +1326,15 @@ public function testGetOptions(): void {
     $this->assertContains('Alaska', $options['values']);
   }
 
+  public function testGetOptionsWithCustom(): void {
+    $this->createCustomGroupWithFieldOfType(['extends' => $this->entity], 'select', 'foo');
+    $this->callAPISuccess('CustomField', 'create', ['id' => $this->ids['CustomField']['fooselect'], 'is_active' => 0]);
+    $options = $this->callAPISuccess($this->entity, 'getoptions', ['field' => 'custom_' . $this->ids['CustomField']['fooselect']]);
+    $this->callAPISuccess('CustomField', 'create', ['id' => $this->ids['CustomField']['fooselect'], 'is_active' => 1]);
+    $options = $this->callAPISuccess($this->entity, 'getoptions', ['field' => 'custom_' . $this->ids['CustomField']['fooselect']]);
```

Comment:
Yea, I don't get it. Why would this api call ever hit this line:

https://github.com/civicrm/civicrm-core/blob/be7f78b6815bbf0a0b41edac36080cf5d399e60a/CRM/Core/DAO.php#L2939-L2942

That line is supposed to be a fallback for extremely weird fields like those added via `fields_callback`... not vanilla custom fields.

Reply:
Does entity->getField return true for disabled fields? I followed where step-debugging took me.

Reply:
My guess @colemanw is that for custom fields when passs in with custom_<x> the fieldName variable is modified in L2932 and as 2940 is an IF not an elseif because it needs to be the fieldName is now stuffed in a PseudoConstant:::get

Reply:
My other thought is does entity->getField deal with customFields at all?

Reply:
or well i suppose the other thing is its a cache problem

Reply:
Possibly but it happens in real life too so it comes up across page requests. But I'm open to it being 5 or 6 problems all combined.

Reply:
which is all probably a turtle somewhere right?